rust-runtime/aws-smithy-runtime/src/client/orchestrator/auth.rs

```diff
@@ -24,33 +24,209 @@ pub(super) async fn orchestrate_auth(
         .map_err(construction_failure)?;
     let identity_resolvers = cfg.identity_resolvers();
 
+    tracing::trace!(
+        auth_option_resolver_params = ?params,
+        auth_options = ?auth_options,
+        identity_resolvers = ?identity_resolvers,
+        "orchestrating auth",
+    );
     for option in auth_options.as_ref() {
         let scheme_id = option.scheme_id();
         let scheme_properties = option.properties();
         if let Some(auth_scheme) = cfg.http_auth_schemes().scheme(scheme_id) {
```

Comment:
In the interest of reducing nesting, we could switch this to something like:

```
let auth_scheme = cfg.http_auth_schemes().scheme(scheme_id).ok_or_else(|| construction_failure(
        "no auth scheme matched auth options. This is a bug. Please file an issue.",
    ))?;
```

Then, `return dispatch_phase.include_mut` below could just be a regular implicit return.

Reply:
I would need to duplicate that error if I did it that way since it's needed for both `if lets`